Sources/XCTAssertAutolayout/Core/CFunctionInjector.swift

```diff
@@ -52,40 +56,75 @@ class CFunctionInjector {
         }
         
         let start = Int(bitPattern: target)
-        let end = start + 2
+        let end = start + 3 * 8
         let pageStart = start & -pageSize
-        let status = mprotect(UnsafeMutableRawPointer(bitPattern: pageStart),
-                              end - pageStart,
-                              PROT_READ | PROT_WRITE | PROT_EXEC)
-        if status == -1 {
-            throw Error(message: "failed to change memory protection: errno=\(errno)")
-        }
+        let size = end - pageStart
+        self.textRange = (UnsafeMutableRawPointer(bitPattern: pageStart), size)
         self.originalFunctionPointer0 = target.assumingMemoryBound(to: Int64.self)
         self.escapedInstructionBytes0 = originalFunctionPointer0.pointee
         self.originalFunctionPointer8 = UnsafeMutablePointer(bitPattern: Int(bitPattern: target) + 8)!
         self.escapedInstructionBytes8 = originalFunctionPointer8.pointee
+        self.originalFunctionPointer16 = UnsafeMutablePointer(bitPattern: Int(bitPattern: target) + 16)!
+        self.escapedInstructionBytes16 = originalFunctionPointer16.pointee
     }
     
     deinit {
         reset()
     }
     
+    /// Disable EXEC and write TEXT segment memory, then enable EXEC again.
+    /// Ref: https://developer.apple.com/documentation/apple-silicon/porting-just-in-time-compilers-to-apple-silicon
+    private func writeText(_ writer: () -> Void) {
+        var status = mprotect(textRange.start, textRange.size, PROT_READ | PROT_WRITE)
+        if status == -1 {
+            fatalError("failed to add write flag to memory protection: errno=\(errno)")
+        }
+        writer()
+        status = mprotect(textRange.start, textRange.size, PROT_READ | PROT_EXEC)
+        if status == -1 {
+            fatalError("failed to add exec flag to memory protection: errno=\(errno)")
```

Comment:
Please replace all calls to fatalError in this function with throwing Error.
Then, when using this function in `inject`, `reset` function, call it with try!, and in `init` check if it is possible to call `mprotect`.

Reply:
OK, I addressed it 👍